streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorNodeTest.java

```diff
@@ -141,4 +154,76 @@ public void testMetrics() {
                 groupName, context.taskId().toString(), node.name())));
     }
 
+    @Test(expected = StreamsException.class)
```

Comment:
As we catch the exception we should swallow is and don't expect it

Reply:
But the catch is only there to test the message content. If the exception is incorrectly not thrown, the test would incorrectly pass. 
Ah, I see below, you're saying the preferred style is to use using assertThrows(), unless the exception is not needed to be caught explicitly in the test?